python/paddle/fluid/tests/demo/pyreader.py

```diff
@@ -0,0 +1,117 @@
+# Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import paddle.fluid as fluid
+import paddle.dataset.mnist as mnist
+import paddle
+import threading
+import numpy
+
+
+def network(is_train):
+    reader, queue = fluid.layers.py_reader(
+        capacity=10,
+        shapes=((-1, 784), (-1, 1)),
+        dtypes=('float32', 'int64'),
+        name="train_reader" if is_train else "test_reader")
+    img, label = fluid.layers.read_file(fluid.layers.double_buffer(reader))
+
+    hidden = img
+
+    for i in xrange(2):
+        hidden = fluid.layers.fc(input=hidden, size=100, act='tanh')
+        hidden = fluid.layers.dropout(
+            hidden, dropout_prob=0.5, is_test=not is_train)
+
+    prediction = fluid.layers.fc(input=hidden, size=10, act='softmax')
+    loss = fluid.layers.cross_entropy(input=prediction, label=label)
+    return fluid.layers.mean(loss), queue, reader
+
+
+def pipe_reader_to_queue(reader_creator, queue):
+    with fluid.program_guard(fluid.Program(), fluid.Program()):
+        feeder = fluid.DataFeeder(
+            feed_list=[
+                fluid.layers.data(
+                    name='img', dtype='float32', shape=[784]),
+                fluid.layers.data(
+                    name='label', dtype='int64', shape=[1])
+            ],
+            place=fluid.CPUPlace())
```

Comment:
It is very noising that we should create data feeder outside.

Maybe we could just monkey patch `reader` instance, and invoke `reader.decorate_python_reader()` to create a thread, and piping data.